corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -420,6 +428,17 @@ hqDefine("data_dictionary/js/data_dictionary", [
             self.removefhirResourceType(false);
         };
 
+        self.toggleShowDeprecatedCaseTypes = function () {
+            self.showDeprecatedCaseTypes(!self.showDeprecatedCaseTypes());
+            const pageUrl = new URL(window.location.href);
+            if (self.showDeprecatedCaseTypes()) {
+                pageUrl.searchParams.append('load_deprecated_case_types', true);
```

Comment:
is it necessary to add `load_deprecated_case_types` to the searchParams in the url if you're adding it as a param on line 252?

Reply:
The reason I kept this solution was because I struggled to find a more elegant way of easily persisting the filter state after a page refresh. Ideally, it would be fantastic if we can remove having it in the URL and so I'm open to suggestions. :thinking:

Reply:
I see. I'm not familiar with the interplay between the window.href url and the one used to retrieve the data from the BE, but my question boils down to: Does window.href have an influence on the URL used to retrieve the data from the BE?

Reply:
They are similar in that the `window.href` query param essentially gets passed to the BE request. When we click the toggle button, a query param gets put into `window.href` and the page refreshes. After the page refresh, we look if the query param exists and then pass that result to the BE request for fetching the JSON data.

Reply:
I see, so it is ultimately needed. Thanks for the clarification

Reply:
It unfortunately seems like it. Alternatively, we could simply store this filter value in the browser cache, but I don't know if that might just complicate things unnecessarily. :man_shrugging:

Reply:
Should be fine keeping it this way. If I'm not mistaken, Case List filters works the same way